tests/api.rs

```diff
@@ -1,36 +1,48 @@
 #[test]
+#[cfg_attr(not(feature = "cargo-clippy"), allow(unknown_lints))]
+#[allow(trivial_regex, string_lit_as_bytes)]
```

Comment:
No, I'm sorry, I cannot abide peppering attributes everywhere for Clippy. These have to go. If that means we don't get clean Clippy output, then we'll just have to live with that.